### PR TITLE
Global Ingestion Lock

### PR DESCRIPTION
**Pull request by nspool**
- add global lock table to schema
- add global lock service
- wrap validation and ingest with lock
- remove all locks on application startup
- web: display an error if request fails

### CODE REVIEW

File: api/src/main/java/au/org/aodn/nrmn/restapi/service/GlobalLockService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import au.org.aodn.nrmn.restapi.data.model.GlobalLock;
+import au.org.aodn.nrmn.restapi.data.repository.GlobalLockRepository;
+
+@Service
+public class GlobalLockService {
+
+    @Autowired 
+    GlobalLockRepository globalLockRepository;
+
```
Comment (utas-raymondng) [inline api/src/main/java/au/org/aodn/nrmn/restapi/service/GlobalLockService.java:14]: Need to use transaction around this section to make read consistency?

**Review by utas-raymondng — CHANGES_REQUESTED** [review]
May need @Transcation to protect the lock

File: api/src/main/java/au/org/aodn/nrmn/restapi/service/GlobalLockService.java (hunk 1 above)
Comment (nspool) [reply api/src/main/java/au/org/aodn/nrmn/restapi/service/GlobalLockService.java:14]: The save in the GlobalLockService is where the lock is acquired, and this method is atomic.
I'll add a `@Transactional` to the setLock method, but in practice it shouldn't matter.

**Review by utas-raymondng — APPROVED** [review]
LGTM

**Pull request merged**